pennylane/qchem/vibrational/bosonic_mapping.py

```diff
@@ -0,0 +1,167 @@
+
+
+def unary_mapping(
+    bose_operator: Union[BoseWord, BoseSentence],
+    nstates: int = 2,
+    ps: bool = False,
+    wire_map: dict = None,
+    tol: float = None,
+):
+    r"""Convert a bosonic operator to a qubit operator using the unary mapping.
+
+    The mapping procedure is described in this `work <https://www.nature.com/articles/s41534-020-0278-0>`_.
+
+    Args:
+        bose_operator(BoseWord, BoseSentence): the bosonic operator
+        nstates(int): maximum number of states a boson can occupy.
+        ps (bool): whether to return the result as a PauliSentence instead of an
+            operator. Defaults to False.
+        wire_map (dict): a dictionary defining how to map the states of
+            the Bose operator to qubit wires. If None, integers used to
+            label the bosonic states will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = qml.qchem.BoseWord()
+    >>> qml.qchem.unary_mapping(w, nstates=4)
+    0.25 * X(0) @ X(1)
+    + -0.25j * X(0) @ Y(1)
+    + 0.25j * Y(0) @ X(1)
+    + (0.25+0j) * Y(0) @ Y(1)
+    + 0.3535533905932738 * X(1) @ X(2)
+    + -0.3535533905932738j * X(1) @ Y(2)
+    + 0.3535533905932738j * Y(1) @ X(2)
+    + (0.3535533905932738+0j) * Y(1) @ Y(2)
+    + 0.4330127018922193 * X(2) @ X(3)
+    + -0.4330127018922193j * X(2) @ Y(3)
+    + 0.4330127018922193j * Y(2) @ X(3)
+    + (0.4330127018922193+0j) * Y(2) @ Y(3)
+    """
+
+    qubit_operator = _unary_mapping_dispatch(bose_operator, nstates, tol=tol)
+
+    wires = list(bose_operator.wires) or [0]
+    identity_wire = wires[0]
+    if not ps:
+        qubit_operator = qubit_operator.operation(wire_order=[identity_wire])
+
+    if wire_map:
+        return qubit_operator.map_wires(wire_map)
+
+    return qubit_operator
+
+
+@singledispatch
+def _unary_mapping_dispatch(bose_operator, nstates, ps=False, wires_map=None, tol=None):
+    """Dispatches to appropriate function if bose_operator is a BoseWord or BoseSentence."""
+    raise ValueError(f"bose_operator must be a BoseWord or BoseSentence, got: {bose_operator}")
+
+
+@_unary_mapping_dispatch.register
+def _(bose_operator: BoseWord, nstates, tol=None):
+
+    if nstates < 2:
+        raise ValueError(f"Number of bosonic states cannot be less than 2, provided {nstates}.")
+
+    cr_op = np.zeros((nstates, nstates))
+    for i in range(nstates - 1):
+        cr_op[i + 1, i] = np.sqrt(i + 1.0)
+
+    op_mat = {"+": cr_op, "-": cr_op.T}
+
+    qubit_operator = PauliSentence({PauliWord({}): 1.0})
+
+    op_prod = defaultdict(list)
+
+    for (_, b_idx), sign in bose_operator.items():
+        op_prod[b_idx].append(sign)
+
+    for b_idx, signs in op_prod.items():
+        op_mat_prod = np.eye(nstates)
+        for sign in signs:
+            op_mat_prod = np.dot(op_mat_prod, op_mat[sign])
```

Comment:
Not sure what op_mat_prod is. A reference would be useful to have.

Reply:
In the nature paper, you see there is a section on avoiding superfluous terms, where the matrices are multiplied before converting them to Pauli operators. This is done to avoid some extra terms which do not add to the value of the operator but make it expensive to calculate. 
Added a comment in the code.